instance/models/load_balancer.py

```diff
@@ -103,20 +111,16 @@ class LoadBalancingServer(ValidateModelMixin, TimeStampedModel):
     accepts_new_backends = models.BooleanField(default=True)
     # A random postfix appended to the haproxy configuration file names to make collisions
     # impossible.
-    fragment_name_postfix = models.CharField(max_length=8, blank=True)
+    fragment_name_postfix = models.CharField(
+        max_length=8,
+        blank=True,
+        default=functools.partial(generate_fragment_name, length=8),
+    )
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.logger = ModelLoggerAdapter(logger, {'obj': self})
 
-    def set_field_defaults(self):
-        if not self.fragment_name_postfix:
-            # Set a unique fragment_name_postfix to avoid clashes between multiple instance
-            # managers sharing the same load balancer.
```

Comment:
This deleted comment explicitly says "between multiple instance managers sharing the same load balancer." which explains the rationale for this code more clearly than the comments you left in, which just say e.g. "A random postfix appended to the haproxy configuration file names to make collisions impossible."

Please edit the comment above `fragment_name_postfix = models.CharField` and/or the docstring of `generate_fragment_name` to include that more explicit rationale (unless it's no longer relevant).